WordCounter/Screen1/Controller/ViewController.swift

```diff
@@ -6,53 +6,37 @@
 //  Copyright © 2019 anhelina.mamekhina. All rights reserved.
 //
 
-import Foundation
 import UIKit
+import RxSwift
+import RxCocoa
 
-class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, TextProcessingDelegate {
+class ViewController: UIViewController {
     
-    var wordsCount: Dictionary<String, Int> = [:]
     let model = WordsCountModel()
+    let disposeBag = DisposeBag()
+    
     @IBOutlet weak var tableView: UITableView!
     
     override func viewDidLoad() {
         super.viewDidLoad()
         tableView.layer.cornerRadius = 10
-        
-        model.delegate = self
     }
     
     override func viewDidAppear(_ animated: Bool) {
         WaitManager.addSpinner(controller: self)
-        model.getDictOfWordsAndCounts()
+        tableView.dataSource = nil
+        tableView.delegate = nil
+        textProcessed()
     }
     
-    func textProcessed(textDictionary: Dictionary<String, Int>) {
-        wordsCount = textDictionary
-        tableView.reloadData()
+    func textProcessed() {
+        model.getDictOfWordsAndCounts().bind(to: tableView.rx.items(cellIdentifier: "Cell", cellType: ListOfWordsCell.self)) {
+            row, dictOfWordsAndCounts, cell in
+            
+            cell.countLabel.text = "\(dictOfWordsAndCounts.value)"
+            cell.wordLabel.text = "\(dictOfWordsAndCounts.key)"
+            } .disposed(by: disposeBag)
         
         WaitManager.dismissSpinner(controller: self)
```

Comment:
Spinner is dismissed wrongly upon screen load, instead of when the results arrive. Use additional binding on the result of the words calculation.